include/ry/runtime_alloc.hpp

```
@@ -0,0 +1,65 @@
+    if (!p && n > 0) oom_abort(n);
+    return p;
+}
+
+inline char *checked_strdup(const char *s) {
+    if (!s) return nullptr;
+    char *p = strdup(s);
+    if (!p) oom_abort(strlen(s) + 1);
+    return p;
+}
+
+inline char *checked_strndup(const char *s, size_t n) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)